sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,502 @@
+    dif_pinmux_padring_insel_t select);
+
+/**
+ * Sets a connection between a Padring MIO output and peripheral output.
+ *
+ * A Padring MIO output can be connected to any available peripheral output.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param index Padring MIO output index.
+ * @param select Peripheral output to be connected to the Padring MIO output.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_select_result_t dif_pinmux_output_select(
+    const dif_pinmux_t *pinmux, dif_pinmux_index_t index,
+    dif_pinmux_peripheral_outsel_t select);
+
+/**
+ * Pin Multiplexer Padring attribute change error codes.
+ */
+typedef enum dif_pinmux_padring_change_result {
+  kDifPinmuxPadringChangeOk = kDifPinmuxOk,
+  kDifPinmuxPadringChangeError = kDifPinmuxError,
+  kDifPinmuxPadringChangeBadArg = kDifPinmuxBadArg,
+  /**
+   * Peripheral is locked and attributes cannot be changed. Only a read has been
+   * performed and this error is recoverable but any operation that writes to
+   * the registers will fail.
+   */
+  kDifPinmuxPadringChangeLocked,
+  /**
+   * Attribute change failed because it would result in conflicting attributes
+   * with another already enabled attribute. No write to the attribute registers
+   * has been attempted. The change can be re-attempted with different bits.
+   * This could also occur if the attribute register already contained a
+   * conflicting set of attributes (e.g. because they have been written outside
+   * of the DIF with conflicting values).
+   */
+  kDifPinmuxPadringChangeConflict,
+} dif_pinmux_padring_change_result_t;
+
+/**
+ * Writes attributes for a pad.
+ *
+ * This function completely overwrites the previous configuration.
+ *
+ * Not all pads implement all attributes and some combinations of cannot be
+ * enabled together. `dif_pinmux_padring_write_attrs` will check for
+ * conflicting attribute and return an appropriate error. If an attribute is
+ * not supported a subsequent read of this attribute will show it as disabled.
+ * It is the caller's responsibility to check an attribute is supported by
+ * reading back the field with `dif_pinmux_padring_read_attr` after writing
+ * this attribute if the support check is desired.
+ *
+ * Note that for additional attributes as the meaning is implementation defined
+ * `dif_padctrl_write_attr` cannot check for invalid combinations that involve
+ * them.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param type Pad type (MIO or DIO).
+ * @param index Pad index to write the attributes for.
+ * @param attrs Attributes to write. A set bit writes an attribute, a
+ *              clear bit leaves it in its existing state.
+ * @return The result of the operation.
+ */
+dif_pinmux_padring_change_result_t dif_padring_write_attrs(
+    const dif_pinmux_t *pinmux, dif_pinmux_padring_pad_kind_t type,
+    dif_pinmux_index_t index, dif_pinmux_padring_pad_attr_t attrs);
```

Comment:
For consistency it might make sense to make the `index` the second parameter in all functions (i.e. always immediately after the `pinmux` handle). Right now it seems to my eye to jump around a bit but don't change it unless it makes sense to you too.